### PR TITLE
example/foundation for embedding a basic ui

### PR DESCRIPTION
**Pull request by whytheplatypus**

### CODE REVIEW

File: upload-server/internal/appconfig/appconfig.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ type RootResp struct {
 type AppConfig struct {
 
 	// App and for Logger
-	LoggerDebugOn bool `env:"LOGGER_DEBUG_ON"`
+	LoggerDebugOn bool `env:"LOGGER_DEBUG_ON, default=true"`
```
Comment (whytheplatypus) [inline upload-server/internal/appconfig/appconfig.go:39]: This probably got in by accident

File: upload-server/internal/metadata/metadata.go [hunk 2]
```diff
@@ -87,6 +87,18 @@ func (c *ConfigCache) SetConfig(key any, config *validation.ManifestConfig) {
 	c.Store(key, config)
 }
 
+func GetMetadataFields(config *validation.ManifestConfig) []validation.FieldConfig {
```
Comment (whytheplatypus) [inline upload-server/internal/metadata/metadata.go:90]: this is very specific, if this logic is needed put it closer to it's use. Could be a template function, or logic within the UI code

File: upload-server/internal/ui/index-demo.html
Comment (whytheplatypus) [inline upload-server/internal/ui/index-demo.html:1]: can get rid of this now, it lives in git if it needs to be referenced.

File: upload-server/internal/ui/manifest.tmpl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        <title>Manifest</title>
+    </head>
+    <body>
+        <h2>Please fill in the sender manifest form for your file</h2>
+        <form method="POST" action="/upload">
+          {{range .MetadataFields}}
+          <div>
+            <label for="{{.FieldName}}">{{FixNames .FieldName}}</label>
+            {{if not .AllowedValues}}
+              <input id="{{.FieldName}}" name="{{.FieldName}}" type="text" />
```
Comment (whytheplatypus) [inline upload-server/internal/ui/manifest.tmpl:15]: needs the `required` behavior.

File: upload-server/internal/ui/manifest.tmpl [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            {{ end }}
+          </div>
+          {{end}}
+          <div>
+            <label for="data_stream_id">Data Stream ID</label>
+            <input id="data_stream_id" name="data_stream_id" type="text" value="{{.DataStream}}" />
+          </div>
+          <div>
+            <label for="data_stream_route">Data Stream Route</label>
+            <input id="data_stream_route" name="data_stream_route" type="text" value="{{.DataStreamRoute}}" />
```
Comment (whytheplatypus) [inline upload-server/internal/ui/manifest.tmpl:31]: these can be hidden.

File: upload-server/internal/ui/assets/tus.min.js
Comment (whytheplatypus) [inline upload-server/internal/ui/assets/tus.min.js:1]: can remove this now

File: upload-server/internal/metadata/metadata.go (hunk 2 above)
Comment (cfarmer-fearless) [reply upload-server/internal/metadata/metadata.go]: Moved to private func in ui.go

File: upload-server/internal/ui/manifest.tmpl (hunk 3 above)
Comment (cfarmer-fearless) [reply upload-server/internal/ui/manifest.tmpl]: done

File: upload-server/internal/ui/assets/tus.min.js
Comment (cfarmer-fearless) [reply upload-server/internal/ui/assets/tus.min.js:1]: Done

File: upload-server/internal/ui/manifest.tmpl (hunk 4 above)
Comment (cfarmer-fearless) [reply upload-server/internal/ui/manifest.tmpl]: done

**Pull request merged**